In [1]:
import pandas as pd

# Load the Titanic dataset with missing values from the provided URL
url = 'https://raw.githubusercontent.com/Geoyi/Cleaning-Titanic-Data/master/titanic_original.csv'

# Load the dataset into a pandas DataFrame
titanic_data = pd.read_csv(url)

# Get missing data counts for each column
missing_data_counts = titanic_data.isnull().sum()

# Display missing data counts
print(missing_data_counts)

pclass          1
survived        1
name            1
sex             1
age           264
sibsp           1
parch           1
ticket          1
fare            2
cabin        1015
embarked        3
boat          824
body         1189
home.dest     565
dtype: int64


## Project Summary:

In this project, the goal was to find and analyze an amusing, interesting dataset with missing values, accessible online as a CSV file. Initially, we explored various options and settled on the famous Titanic dataset. The dataset contains information about passengers on the Titanic, including attributes such as survival status, age, cabin, and more. Significant amounts of missing data are present in columns like age, cabin, boat, body, and home.dest.

After downloading the dataset, the next step was to load it using Pandas, a Python library, to analyze and count the missing values. We wrote the following code:

```
import pandas as pd

# Load the Titanic dataset with missing values from the provided URL
url = 'https://raw.githubusercontent.com/Geoyi/Cleaning-Titanic-Data/master/titanic_original.csv'

# Load the dataset into a pandas DataFrame
titanic_data = pd.read_csv(url)

# Get missing data counts for each column
missing_data_counts = titanic_data.isnull().sum()

# Display missing data counts
print(missing_data_counts)
```
The dataset revealed extensive missing data, particularly in fields like age (264 missing), cabin (1015 missing), boat (824 missing), and body (1189 missing). This provided a useful exercise in identifying and handling missing data, which is a critical step in data cleaning and preprocessing for analysis.
